exporter/kafkaexporter/config.go

```diff
@@ -41,8 +42,47 @@ type Config struct {
 	// Client, and shared by the Producer/Consumer.
 	Metadata Metadata `mapstructure:"metadata"`
 
-	// TODO authentication
-	// TODO batch settings
+	// Authentication defines used authentication mechanism.
+	Authentication Authentication `mapstructure:"authentication"`
+}
+
+// AuthType defines authentication type
+type AuthType string
+
+const (
+	// No authentication
+	AuthTypeNone AuthType = "none"
+	// Plain text authentication
+	AuthTypePlaintext AuthType = "plain_text"
+	// TLS authentication
+	AuthTypeTLS AuthType = "tls"
+	// Kerberos authentication
+	AuthTypeKerberos AuthType = "kerberos"
+)
+
+// Authentication defines authentication.
+type Authentication struct {
+	Type      AuthType                   `mapstructure:"type"`
```

Comment:
Can we do something similar to Jaeger receiver (see protocols)? instead of asking users to set type + one of the map entry, to just ask user to set one of the possible authentifications?

```
exporters:
  kafka:
    auth:
      tls:
```

OR 

```
exporters:
  kafka:
    auth:
      kerberos:
        ....
```

Reply:
Done